shpc/main/registry/remote.py

```diff
@@ -106,11 +106,18 @@ def web_url(self):
         """
         Retrieve the web url, either pages or (eventually) custom.
         """
-        parts = self.source_url.split("/")[3:]
-        return "https://%s.%s.io/%s/library.json" % (
-            parts[0],
-            self.provider_name,
-            "/".join(parts[1:]),
+        parts = self.source_url.split("/")[2:]
+        domain = parts[0].split(".")
```

Comment:
This is splitting on an unknown number of `.` are you sure you don't want to be specific? 

I think what I would like to see here is parsing the url, and then formally splitting into prefix/suffix, and then using those specifically. The check for if domain == a list is also erroneous and makes assumptions about that structure.

Pinging @muffato for his thoughts since he wrote the original logic.

Reply:
My understanding was that the `parts` split already yields the suffix, gets rid of the `https://` prefix (and what follows is the full domain which can have a variable number of dots)

Reply:
Ok, I changed it to make the parsing more specific.
About the if clause: it is meant to handle the case for git{hub,lab}.com, I think it should safe to make that assumption about the structure there.

Reply:
oh definitely this is much more clear! Have you tested it with your registry and you are happy with it?

If so it is good for me - for the final tweaks please add a note to the CHANGELOG.md with a version bump, and do the same bump in the shpc/version.py file. And after that we will just want @muffato to review and approve too.

Thanks for fixing this!

Reply:
> oh definitely this is much more clear! Have you tested it with your registry and you are happy with it?
works like a charm :rocket: 

> If so it is good for me - for the final tweaks please add a note to the CHANGELOG.md with a version bump, and do the same bump in the shpc/version.py file. And after that we will just want @muffato to review and approve too.

Done :)